keras/backend/theano_backend.py

```diff
@@ -357,8 +357,11 @@ def __call__(self, inputs):
         return self.function(*inputs)
 
 
-def function(inputs, outputs, updates=[]):
-    return Function(inputs, outputs, updates=updates)
+def function(inputs, outputs, updates=[], debug_nan=False):
+    mode = None
+    if _DEBUG_MODE == 'detect_nan':  # use suggested Theano detect_nan implementation
+        mode = theano.compile.MonitorMode(post_func=theano.compile.monitormode.detect_nan).excluding('local_elemwise_fusion', 'inplace')
```

Comment:
Do I understand you correctly that you would rather see line 363 changed with `mode = 'NanGuardMode'`?

Or change it to `if mode and mode != 'none': mode = _DEBUG_MODE` to just make the mode parameter configurable through Keras configuration?

Or do you suggest running with environment variable `THEANO_FLAGS='mode=NanGuardMode'` to let Theano handle such debugging purposes? (Unfortunately this does not work at the moment and it forces the use of GPU, even if you pass `dev=cpu`.)